Prefect Idiom documentation section (#2455)
PrefectHQ/prefect: docs/core/faq/branching.md

@task
def output_value(value):
    print(value)

with Flow("branches") as flow:
    n = get_number()

    m = multiply(n)
    d = divide(n)

    output_value(m)
    output_value(d)
```
:::

::: tab "Imperative API"
```python
from prefect import Task, Flow

class GetNumber(Task):
    def run(self):
        return 10

class Multiply(Task):
    def run(self, n):
        return n * 10

class Divide(Task):
    def run(self, n):
        return n / 10

class OutputValue(Task):
    def run(self, value):
        print(value)

flow = Flow("branches")

n = GetNumber()

m = Multiply()
d = Divide()

m.set_upstream(n, key="n", flow=flow)
d.set_upstream(n, key="n", flow=flow)

output_1 = OutputValue()
output_2 = OutputValue()

output_1.set_upstream(m, key="value", flow=flow)
output_2.set_upstream(d, key="value", flow=flow)
```
:::
::::

### Branching based on conditional logic

Prefect ships with various [tasks for control logic](/core/task_library/control_flow.html) that you can use off-the-shelf to control the branching structure of your flow. This is especially useful in cases where under certain conditions you want to control which branches of your flow run based on upstream events.

> Let's say you want to have a flow where some conditional task is evaluated and then the downstream branches are run depending on the output of that conditional task—true or false. An example of this might look like the following:

Reviewer: ```suggestion
Let's say you want to have a flow where some conditional task is evaluated and then the downstream branches are run depending on the output of that conditional task. An example of this might look like the following:
```

![Flow with Conditional Branching](/faq/conditional_branch.png)

> If you choose to use some of Prefect's default control flow tasks such as `if_else` and `merge` then the flow will take on a DAG with some additional checks added:

Reviewer: ```suggestion
If you choose to use some of Prefect's default control flow tasks such as `if_else` and `merge` then Prefect will automatically add some additional tasks to your Flow:
```
